src/main/java/core/basesyntax/service/operations/PurchaseOperation.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.model.Product;
+import java.util.List;
+
+public class PurchaseOperation implements Operation {
+    @Override
+    public boolean updateStorage(List<Product> products) {
+        List<Product> list = Operation.storageService.getAll();
+        int countToBuy = products.size();
+        long countInStock = Operation.storageService.getAll().stream()
+                .filter(p -> p.getFruitType().equalsIgnoreCase(products.get(0).getFruitType())
+                && p.getExpirationDate().isAfter(products.get(0).getExpirationDate()))
+                .count();
+        if (countInStock - countToBuy < 0) {
+            return false;
+        } else {
+            for (int i = 0; i < list.size(); i++) {
+                if (list.get(i).getFruitType().equalsIgnoreCase(products.get(0).getFruitType())
```

Comment:
complex `if` statement. Can we move this to a separate method, that will return `boolean` as a result of checking?